File: backend/src/main/java/br/com/ifsp/classify/dtos/create/FormQuestionCreateDTO.java
package br.com.ifsp.classify.dtos.create;

import br.com.ifsp.classify.models.form.AnswerType;

import java.util.List;

public record FormQuestionCreateDTO(
        String question,
        AnswerType answerType,
        Boolean isRequired,
        List<FormQuestionOptionCreateDTO> options
) {
}

File: frontend/src/shared/dtos/form-questions/FormQuestionCreateDTO.ts
import type {FormQuestionOptionCreateDTO} from "@/shared/dtos/form-question-options/FormQuestionOptionCreateDTO.ts";
import type {AnswerType} from "@/shared/models/enums/answer-type.ts";

export interface FormQuestionCreateDTO {
    question: string,
    answerType: AnswerType,
    isRequired: boolean,
    options?: FormQuestionOptionCreateDTO[]
}
